From 8011b4936eba4be9056595ee27a0bf0e62526df9 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sat, 12 Jul 2025 22:03:55 +0200
Subject: [PATCH 105/115] arm64: dts: mediatek: mt6785-xiaomi-begonia: Enable
 GPU

Add support for the GPU by enabling gpu node and adding vgpu regulator to it.

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi      | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
index 2887db82a..2045b701e 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
@@ -116,6 +116,11 @@ &dsi0_out {
 	remote-endpoint = <&panel_in>;
 };
 
+&gpu {
+	mali-supply = <&mt6359_vgpu11_buck_reg>;
+	status = "okay";
+};
+
 &i2c0 {
 	clock-frequency = <400000>;
 
-- 
2.56.0

